src/inmanta/data/__init__.py

```diff
@@ -5547,6 +5542,13 @@ class User(BaseDocument):
     password_hash: str
     auth_method: AuthMethod
 
+    @classmethod
+    def table_name(cls) -> str:
+        """
+        Return the name of the user table prepended by "public." to differentiate it from the user table in the pg schema.
+        """
+        return f"public.{cls.__name__.lower()}"
+
```

Comment:
I find this code rather confusing. This is a single method that behaves differently for one specific subclass. Why didn't you choose for the approach where the method is parameterized and the default value is set to False?

Reply:
This hadn't struck me, but I do like the use of polymorphism here.

Reply:
because I caused fall outs at multiple places. For example in the `insert` method:
It calls `self.table_name()` but If I want to specify the parameter to it (which I need for the `insert` method of a User) , it will also become a parameter of the `insert` method which I didn't like.

Reply:
It doesn't have to become a parameter of the `insert` method, does it. Can't you just use `table_name(include_schema=True)` in the insert method?

Reply:
Indeed, I made the change at all the other places it was needed. I personally have no strong preference. Both seem readable to me.

Reply:
@arnaudsjs I do like the use of polymorphism here a lot. 

Because here it is clear why it is different from the others (because of the name of this table).
And it still does what the `table_name` method is expected to do: return the table name to use in queries. 

In the insert method on the parent class, why it needs the prefix is not at all apparent.